source/MaterialXCore/CMakeLists.txt

```diff
@@ -7,6 +7,10 @@ add_library(MaterialXCore ${materialx_source} ${materialx_headers})
 
 add_definitions(-DMATERIALX_CORE_EXPORTS)
 
+if (MATERIALX_BUILD_SHARED_LIBS)
+target_compile_definitions(MaterialXCore PUBLIC -DMATERIALX_BUILD_SHARED_LIBS)
+endif()
+
```

Comment:
Then it needs to expose to the public the need to define MATERIALX_BUILD_SHARED_LIBS.